git_ui: Improve error handling in commit message generation

imumesh18 opened the pull request:
After and Before video of the issue and solution.

https://github.com/user-attachments/assets/40508f20-5549-4b3d-9331-85b8192a5b5a



Closes #27319

Release Notes:

- Provide Feedback on commit message generation error

imumesh18 committed:
Improve error handling in commit message generation

Signed-off-by: Umesh Yadav <umesh4257@gmail.com>

imumesh18 committed:
Merge branch 'main' into commit-msg-error

imumesh18 committed:
fix errors

imumesh18 committed:
📎

imumesh18 committed:
Merge branch 'main' into commit-msg-error

imumesh18 committed:
Merge branch 'main' into commit-msg-error

imumesh18 committed:
Merge branch 'main' into commit-msg-error

imumesh18 committed:
Merge branch 'main' into commit-msg-error

cole-miller commented:
Thanks for working on this and sorry for the delay in reviewing. This looks good, my only feedback is that I think we should use a normal toast (the kind that appears in the bottom-right corner of the screen) instead of the custom toast we use for git remote operations---the click-through to a buffer with the very short error message for this case isn't a great experience. Do you think you could update the code to use that approach?

imumesh18 commented:
Hey @cole-miller yes. I will update it and get back to you.

imumesh18 committed:
show commit errors via workspace API

imumesh18 committed:
Merge branch 'main' into commit-msg-error

imumesh18 committed:
Refactor commit message error handling into helper method

imumesh18 commented:
Hey @cole-miller , I have updated the code and modified the error toast. For now I have added the message error context like Failed to generate commit message in the notification body as I couldn't find a way to add that text as title. Let me know if you think this can be done better.

<img width="504" alt="image" src="https://github.com/user-attachments/assets/86017a38-0b80-49d3-bbc3-c5392165b23a" />

cole-miller committed:
Merge remote-tracking branch 'origin/main' into commit-msg-error

cole-miller reviewed (approved):
Thanks @imumesh18!
